base_name_search_improved/models/ir_model.py

```diff
@@ -65,19 +156,111 @@ def name_search(self, name='', args=None,
                         domain = [(rec_name, operator, name.replace(' ', '%'))]
                         res = _extend_name_results(
                             self, base_domain + domain, res, limit)
+
                     # Try unordered word search on each of the search fields
-                    for rec_name in all_names:
-                        domain = [(rec_name, operator, x)
-                                  for x in name.split() if x]
+                    # we only perform this search if we have at least one
+                    # separator character
+                    separator = _get_separator(self)
+                    if separator in name:
+                        domain = []
+                        for word in name.split(separator):
+                            word_domain = []
+                            for rec_name in all_names:
+                                word_domain = (
```

Comment:
Could you help us with a comment to explain this piece of code (I didn't understand it, sorry)?